src/db/collections/block.rs

```diff
@@ -132,64 +131,47 @@ pub struct IncludedBlockResult {
 
 #[derive(Deserialize, Debug, Clone)]
 pub struct IncludedBlockMetadataResult {
+    #[serde(rename = "_id")]
     pub block_id: BlockId,
     pub metadata: BlockMetadata,
 }
 
+#[derive(Deserialize)]
+struct RawResult {
+    #[serde(with = "serde_bytes")]
+    raw: Vec<u8>,
+}
+
+#[derive(Deserialize)]
+struct BlockIdResult {
+    #[serde(rename = "_id")]
+    block_id: BlockId,
+}
+
 /// Implements the queries for the core API.
 impl BlockCollection {
     /// Get a [`Block`] by its [`BlockId`].
     pub async fn get_block(&self, block_id: &BlockId) -> Result<Option<Block>, Error> {
-        self.aggregate(
-            [
-                doc! { "$match": { "_id": block_id } },
-                doc! { "$lookup": {
-                    "from": OutputCollection::NAME,
-                    "localField": "_id",
-                    "foreignField": "metadata.block_id",
-                    "pipeline": [
-                        { "$sort": { "_id": 1 } },
-                        { "$replaceWith": "$output" }
-                    ],
-                    "as": "block.payload.essence.outputs"
-                } },
-                // Stupidly, if the block has no payload, then the above lookup
-                // will add the structure, causing the deserialization to fail.
-                // So this is needed to make sure we remove it if necessary.
-                doc! { "$set": { "block.payload": { "$cond": [
-                    { "$not": [ "$block.payload.kind" ] },
-                    "$$REMOVE",
-                    "$block.payload",
-                ] } } },
-                doc! { "$replaceWith": "$block" },
-            ],
-            None,
-        )
-        .await?
-        .try_next()
-        .await
+        Ok(self
+            .get_block_raw(block_id)
+            .await?
+            .map(|raw| iota_types::block::Block::unpack_unverified(raw).unwrap().into()))
```

Comment:
I have a very bad feeling about this line.

Reply:
How so?

Reply:
Well ... what happens if the binary representation changes in the future? How do we then unpack old Blocks if we do it unverified?

Reply:
If the representation changes, there will have to be a version change in `iota_types`, and most likely there will have to be a solution in that crate for old representations.

Reply:
I am unsure here. Arent' the protocol parameters all it needs as context to unpack older versions of blocks?

Reply:
right, thx